ckan/logic/action/update.py

```diff
@@ -93,6 +94,10 @@ def resource_update(context, data_dict):
         plugin.before_update(context, pkg_dict['resources'][n], data_dict)
 
     upload = uploader.get_resource_uploader(data_dict)
+    upload.upload(id, uploader.get_max_resource_size())
+
+    data_dict['mimetype'] = mimetypes.guess_type(data_dict['url'])[0] or upload.mimetype
```

Comment:
Same here, if the user provides a value for `mimetype` or `size`, don't override it